src/lib/auth.ts

```diff
@@ -0,0 +1,42 @@
+import { atom, selector, setState } from "atom.io"
+import * as FirebaseAuth from "firebase/auth"
+
+import type { SystemUser } from "../types"
+import { auth } from "./firebase"
+import { systemUserAtoms } from "./users"
+
+export type AuthStatus =
+	| { authenticated: false; loading: boolean }
+	| { authenticated: true; loading: false; me: FirebaseAuth.User }
+export const myAuthStatusAtom = atom<AuthStatus>({
+	key: `myAuthStatus`,
+	default: { authenticated: false, loading: true },
+	effects: [
+		({ setSelf }) => {
+			const unSub = FirebaseAuth.onAuthStateChanged(auth, (myFirebaseUser) => {
+				if (myFirebaseUser) {
+					setSelf({ authenticated: true, loading: false, me: myFirebaseUser })
+				} else {
+					setSelf({ authenticated: false, loading: false })
+				}
+			})
+			return unSub
+		},
+	],
+})
+export const myselfSelector = selector<SystemUser | null>({
+	key: `myself`,
+	get: ({ get }) => {
+		const myAuthStatus = get(myAuthStatusAtom)
+		if (myAuthStatus.authenticated) {
+			const myself = get(systemUserAtoms, myAuthStatus.me.uid)
+			return myself
+		}
+		return null
+	},
+})
+export function logout(): void {
+	void FirebaseAuth.signOut(auth).then(() => {
+		setState(myAuthStatusAtom, { authenticated: false, loading: false })
+	})
```

Comment:
no `.catch()` after `.then()` — probably okay, but I always like to see a catch so they're not dangling